### PR TITLE
Incorporate unit tests for action execution core component 

### PR DESCRIPTION
**Pull request by malithie**
### Proposed changes in this pull request

This PR adds unit tests covering core functionality of the action execution core component introduced along with https://github.com/wso2/product-is/issues/20739

With this PR following coverage targets are achieved.
- Complexity (branches, conditions, etc.) -> 0.62
- Line (percentage of executed lines in the source code) -> 0.8

### When should this PR be merged

soon

### Follow up actions

NA

### CODE REVIEW

File: components/action-mgt/org.wso2.carbon.identity.action.execution/src/test/java/org/wso2/carbon/identity/action/execution/ActionExecutorServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,438 @@
...
+
+    @Test(expectedExceptions = ActionExecutionException.class,
+            expectedExceptionsMessageRegExp = "Multiple actions found for action type: PRE_ISSUE_ACCESS_TOKEN. " +
+                    "Current implementation doesn't support multiple actions for a single action type.")
+    public void testActionExecuteFailureWhenMultipleActionsAvailableForActionType() throws Exception {
+
+        List<Action> mockActions = Arrays.asList(mock(Action.class), mock(Action.class));
+        when(actionManagementService.getActionsByActionType(any(), any())).thenReturn(mockActions);
+
+        actionExecutorService.execute(ActionType.PRE_ISSUE_ACCESS_TOKEN, any(), any());
```
Comment (Thisara-Welmilla) [inline components/action-mgt/org.wso2.carbon.identity.action.execution/src/test/java/org/wso2/carbon/identity/action/execution/ActionExecutorServiceImplTest.java:129]: Shall we add assert for actionExecutionStatus

File: components/action-mgt/org.wso2.carbon.identity.action.execution/src/test/java/org/wso2/carbon/identity/action/execution/ActionExecutorServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,438 @@
...
+    }
+
+    @Test(expectedExceptions = ActionExecutionException.class,
+            expectedExceptionsMessageRegExp = "No request builder found for action type: PRE_ISSUE_ACCESS_TOKEN")
+    public void testActionExecuteFailureWhenNoRegisteredRequestBuilderForActionType() throws Exception {
+
+        when(actionManagementService.getActionsByActionType(any(), any())).thenReturn(
+                Collections.singletonList(mock(Action.class)));
+
+        actionExecutorService.execute(ActionType.PRE_ISSUE_ACCESS_TOKEN, any(), any());
```
Comment (Thisara-Welmilla) [inline components/action-mgt/org.wso2.carbon.identity.action.execution/src/test/java/org/wso2/carbon/identity/action/execution/ActionExecutorServiceImplTest.java:139]: Shall we add an assert here for actionExecutionStatus. Check other places as well.

File: components/action-mgt/org.wso2.carbon.identity.action.execution/src/test/java/org/wso2/carbon/identity/action/execution/util/APIClientTest.java [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+        when(httpResponse.getStatusLine()).thenReturn(statusLine);
+        when(statusLine.getStatusCode()).thenReturn(HttpStatus.SC_OK);
+        InputStreamEntity entity =
+                new InputStreamEntity(new ByteArrayInputStream("{}".getBytes(StandardCharsets.UTF_8)));
+        entity.setContentType(ContentType.DEFAULT_TEXT.getMimeType());
+        when(httpResponse.getEntity()).thenReturn(entity);
+
+        ActionInvocationResponse apiResponse = apiClient.callAPI("http://example.com", null, "{}");
+        assertNotNull(apiResponse);
+        assertNull(apiResponse.getResponse());
```
Comment (Thisara-Welmilla) [inline components/action-mgt/org.wso2.carbon.identity.action.execution/src/test/java/org/wso2/carbon/identity/action/execution/util/APIClientTest.java:93]: Shall we add the assert for the ActionInvocationResponse.actionStatus as well.

File: components/action-mgt/org.wso2.carbon.identity.action.execution/src/test/java/org/wso2/carbon/identity/action/execution/ActionExecutorServiceImplTest.java (hunk 1 above)
Comment (malithie) [reply components/action-mgt/org.wso2.carbon.identity.action.execution/src/test/java/org/wso2/carbon/identity/action/execution/ActionExecutorServiceImplTest.java:129]: What's expected here is the exception. So there's really no need to assert the status

File: components/action-mgt/org.wso2.carbon.identity.action.execution/src/test/java/org/wso2/carbon/identity/action/execution/ActionExecutorServiceImplTest.java (hunk 2 above)
Comment (malithie) [reply components/action-mgt/org.wso2.carbon.identity.action.execution/src/test/java/org/wso2/carbon/identity/action/execution/ActionExecutorServiceImplTest.java:139]: What's expected here is the exception. So there's really no need to assert the status

**Review by Thisara-Welmilla — APPROVED** [review]

**Pull request merged**